controllers/create_apideployment.go

```diff
@@ -0,0 +1,136 @@
+	deploy *appsv1.Deployment,
+	instance *cachev1.OCMSearch,
+) (*reconcile.Result, error) {
+
+	found := &appsv1.Deployment{}
+	err := r.Get(context.TODO(), types.NamespacedName{
+		Name:      deploy.Name,
+		Namespace: request.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+
+		err = r.Create(context.TODO(), deploy)
+		if err != nil {
+			return &reconcile.Result{}, err
+		} else {
+			return nil, nil
+		}
+	} else if err != nil {
+		return &reconcile.Result{}, err
+	}
+
+	return nil, nil
+}
+
+func (r *OCMSearchReconciler) APIDeployment(instance *cachev1.OCMSearch) *appsv1.Deployment {
+
+	image_sha := os.Getenv("API_IMAGE")
+	log.V(2).Info("Using api image ", image_sha)
+	deploymentLabels := map[string]string{
+		"name": "search-api",
+	}
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "search-api",
+			Namespace: instance.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: deploymentLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: deploymentLabels,
+				},
+			},
+		},
+	}
+	indexerContainer := corev1.Container{
+		Name: "search-api",
+		//Image: "registry.redhat.io/rhscl/postgresql-13-rhel7:1-31.1638430360",
+		Image: image_sha,
+		Env: []corev1.EnvVar{
+			newSecretEnvVar("DB_USER", "database-user", "search-postgres"),
+			newSecretEnvVar("DB_PASS", "database-password", "search-postgres"),
+			newSecretEnvVar("DB_NAME", "database-name", "search-postgres"),
+			newEnvVar("DB_HOST", "search-postgres."+instance.Namespace+".svc"),
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "search-api-certs",
+				MountPath: "/sslcert",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("20m"),
+				corev1.ResourceMemory: resource.MustParse("100Mi"),
+			},
+		},
+		ReadinessProbe: &corev1.Probe{
+			InitialDelaySeconds: 5,
+			TimeoutSeconds:      1,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Port:   intstr.FromInt(4010),
+					Path:   "/readiness",
+					Scheme: "HTTPS",
+				},
+			},
+		},
+		LivenessProbe: &corev1.Probe{
+			InitialDelaySeconds: 30,
+			TimeoutSeconds:      1,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Port:   intstr.FromInt(4010),
+					Path:   "/liveness",
+					Scheme: "HTTPS",
+				},
+			},
+		},
+	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: "search-api-certs",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "search-api-certs",
+				},
+			},
+		},
+	}
+	var replicas int32 = 1
+	deployment.Spec.Replicas = &replicas
+
+	deployment.Spec.Template.Spec.Containers = []corev1.Container{indexerContainer}
+	deployment.Spec.Template.Spec.Volumes = volumes
+	deployment.Spec.Template.Spec.ServiceAccountName = "search-v2-operator"
```

Comment:
Remove `v2`

Reply:
@jlpadilla can that be part of name update , coming up .

Reply:
ok, we can do the rename in a separate PR.